<launch>
    <arg name="interactive_manipulation" default="true"/>
    <arg name="sim" default="false"/>
<!--<<<<<<< HEAD
    <arg name="standalone" default="false"/>
    <arg name="use_right_arm" default="true"/>
    <arg name="use_left_arm" default="true"/>

    <include if="$(arg interactive_manipulation)" file="$(find pr2_interactive_manipulation)/launch/pr2_interactive_manipulation_robot.launch"/>
    <group unless="$(arg interactive_manipulation)">
        <include file="$(find pr2_object_manipulation_launch)/launch/c_jt.launch">
            <arg if="$(arg use_right_arm)" name="use_right_arm" value="true" />
            <arg if="$(arg use_left_arm)"  name="use_left_arm" value="true" />
        </include>

        <include if="$(arg use_right_arm)" file="$(find pr2_arm_navigation_kinematics)/launch/right_arm_collision_free_ik.launch"/>
        <include if="$(arg use_left_arm)"  file="$(find pr2_arm_navigation_kinematics)/launch/left_arm_collision_free_ik.launch"/>
        <param name="pr2_right_arm_kinematics/object_padd" value="0.04"/>
        <param name="pr2_left_arm_kinematics/object_padd" value="0.04"/>
    </group>
=======
-->
    <arg name="interactive_manipulation" default="true"/>
    <arg name="arserver" default="false"/>
    <arg name="maps" default="true"/>
    <arg name="rosbridge" default="true"/>

    <arg name="wg" default="false"/>
    <arg name="aware_home" default="false"/>
    <arg name="hrl" default="true"/>
    <arg name="henry" default="false"/>

    <machine name="c1" address="c1" default="true"/>
    <machine name="c2" address="c2" env-loader="/u/cbsdemo/env.sh"/>


    <group if="$(arg maps)">
        <group if="$(arg aware_home)">
            <node name="$(anon mapserver)" pkg="map_server" type="map_server" args="$(find gt_pr2)/navigation/maps/aware_home_025.yaml"/>
        </group>

        <group if="$(arg wg)">
          <node name="map_server" pkg="map_server" type="map_server"
                args="$(find willow_maps)/willow-sans-whitelab-2010-02-18-0.025.pgm 0.025" machine="c2" />
        </group>

        <group if="$(arg henry)">
          <node name="map_server" pkg="map_server" type="map_server" args="$(find henry_nav)/map_0-25/map.pgm 0.025" />
        </group>

        <group if="$(arg hrl)">
            <node name="map_server" pkg="map_server" type="map_server" args="$(find gt_pr2)/navigation/maps/hsi_full_4_24.yaml" machine="c2"/>
        </group>
    </group>

    <include if="$(arg interactive_manipulation)" file="$(find pr2_interactive_manipulation)/launch/pr2_interactive_manipulation_robot.launch">
        <arg name="nav" value="true"/>
    </include>

    <include unless="$(arg sim)" file="$(find tts_server)/tts_server.launch"/>

    <include unless="$(arg sim)" file="$(find rcommander_pr2_gui)/launch/pr2_gripper_sensor_actions.launch"/>

    <!--<include file="$(find face_detector)/launch/face_detector_action.wide.launch"/>-->

    <!--c2-->
    <include file="$(find rcommander_pr2_gui)/launch/ptp_arm_action.launch"/>
<!--<<<<<<< HEAD
    <node pkg="rcommander_pr2_gui" type="rcommander_tuckarm_node.py" output="screen" name="rc_tuck_arms_node"/>
    <node pkg="tf_broadcast_server" type="tf_broadcast_server.py" output="screen" name="rcommander_tf_broadcast"/>

    <node unless="$(arg standalone)" pkg="rcommander_pr2_gui" type="min_time_node" output="screen" name="min_time_to_move_node"/>
    <node unless="$(arg standalone)" pkg="topic_tools" type="throttle" output="screen" name="joint_states_throttler" args="messages joint_states 20."/>
=======
-->

    <!--c2-->
    <include file="$(find rcommander_ar_pose)/launch/ar_pose.launch">
         <arg name="kinect" value="true"/>
    </include>

    <!--c2-->
    <include if="$(arg rosbridge)" file="$(find rcommander_web)/launch/rosbridge.launch"/>

    <include file="$(find rcommander_pr2_gui)/launch/rcommander_robot_brittle.launch"/>

    <group if="$(arg arserver)">
        <node pkg="rcommander_pr2_gui" type="rcommander_pr2_ar_server.py" 
            output="screen" name="rcommander_pr2_ar_server"
            args="$(env HOME)/action_database.pkl $(env HOME)/tag_database.pkl  $(env HOME)/robot_behaviors">
        </node>
    </group>
</launch>
